### PR TITLE
Guil8553/5362 remove qml

### PR DESCRIPTION
**Pull request by GuillaumeBelz**
Most of changes are just renaming.

I needed to do a lot of commits because there was a problem with validation script (fixed now)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -21,10 +21,9 @@
     "Scalebar",
     "Struct",
     "subclassing",
-    "toolkitcpp",
-    "toolkitqml",
+    "toolkit",
```
Comment (GuillaumeBelz) [inline .vscode/settings.json:24]: I'm not sure how this JSON is used. Maybe it's better to keep "toolkitcpp"? @jared-2016

File: augmented_reality/ArApi/ArApi.pri [hunk 2]
```diff
@@ -0,0 +1,110 @@
+###############################################################################
```
Comment (GuillaumeBelz) [inline augmented_reality/ArApi/ArApi.pri:1]: I merged common and api project files and tried to simplified architecture for AR API

File: uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/SearchViewController.cpp [hunk 3]
```diff
@@ -685,7 +685,7 @@ namespace Esri::ArcGISRuntime::Toolkit {
                   else
                   {
                     // Take ownership of all elements.
-                    for (auto r : qAsConst(results))
```
Comment (GuillaumeBelz) [inline uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/SearchViewController.cpp:688]: qAsConst will be deprecated

File: uitools/unit_tests/BasemapGalleryItemUnitTest/tst_basemapgalleryitemunittest.cpp [hunk 4]
```diff
@@ -19,33 +19,30 @@
 #include "AutoDisconnector.h"
 #include "Basemap.h"
 #include "BasemapGalleryItem.h"
+#include "Error.h"
```
Comment (GuillaumeBelz) [inline uitools/unit_tests/BasemapGalleryItemUnitTest/tst_basemapgalleryitemunittest.cpp:22]: It looks like unit tests were not updated for long time. Fixed. Maybe we should add them in GitHub action.

File: .vscode/settings.json (hunk 1 above)
Comment (jared-2016) [reply .vscode/settings.json:24]: quick google, looks like project specific configuration that adds these items to the spellcheck. Im guessing so these items don't get the red squiggly line underneath them. Maybe do a quick search of the toolkit repo with `toolkitcpp` and see if there are a lot of hits and decide from there?

File: calcite/demo/main.cpp [hunk 5]
```diff
@@ -41,7 +41,7 @@ main(int argc, char* argv[])
   // (https://links.esri.com/arcgis-api-keys).
 
   const QString apiKey =
-    QString("");
+      QString("");
```
Comment (jared-2016) [inline calcite/demo/main.cpp:44]: is this spacing correct?

Comment (GuillaumeBelz) [reply calcite/demo/main.cpp:44]: maybe no. I guess it was the default indentation of Qt Creator. I guess it's better to remove the new line.

Comment (GuillaumeBelz) [reply calcite/demo/main.cpp:44]: Oh, I forgot it's calcite. I tried to fix it in first time, but it's too complicated and I saw there is an issue for calcite. I canceled my changes.

File: .vscode/settings.json (hunk 1 above)
Comment (jared-2016) [reply .vscode/settings.json:24]: After a closer look I see that `toolkitcpp` is still in use, we should probably leave that one in until we remove it's usage from the repo.

**Comment by GuillaumeBelz** [comment]
I just saw I missed a lot of changes in documentations

File: uitools/unit_tests/BasemapGalleryItemUnitTest/tst_basemapgalleryitemunittest.cpp (hunk 4 above)
Comment (jared-2016) [reply uitools/unit_tests/BasemapGalleryItemUnitTest/tst_basemapgalleryitemunittest.cpp:22]: I like this idea.

**Review by jared-2016 — COMMENTED** [review]
I was taking a quick look over the repo and came across [toolkitqml.pri](https://github.com/Esri/arcgis-maps-sdk-toolkit-qt/blob/guil8553/5362-remove-qml/uitools/toolkitcpp/toolkitqml.pri). I don't think this is needed anymore.

File: calcite-qml/demo/arcgisruntime.pri [hunk 6]
```diff
@@ -25,6 +25,6 @@ iniLine = $$cat($${iniPath}, "lines")
 dirPath = $$find(iniLine, "InstallDir")
 cleanDirPath = $$replace(dirPath, "InstallDir=", "")
 priLocation = $$replace(cleanDirPath, '"', "")
-!include($$priLocation/sdk/ideintegration/arcgis_runtime_qml.pri) {
```
Comment (GuillaumeBelz) [inline calcite-qml/demo/arcgisruntime.pri:28]: @jared-2016 I get your last changes from v.next, I don't know why these changes are detected in GitHub.

Comment (GuillaumeBelz) [reply calcite-qml/demo/arcgisruntime.pri:28]: Why `calcite-qml` and not just `calcite`?

Comment (jared-2016) [reply calcite-qml/demo/arcgisruntime.pri:28]: Oh the folder containing calcite demo? it just slipped my radar. You can put that change in this PR. Sorry for that.

Comment (jared-2016) [reply calcite-qml/demo/arcgisruntime.pri:28]: > @jared-2016 I get your last changes from v.next, I don't know why these changes are detected in GitHub.

Yeah that's interesting. This was what I changed. 🤷

Comment (GuillaumeBelz) [reply calcite-qml/demo/arcgisruntime.pri:28]: Done

**Comment by GuillaumeBelz** [comment]
For a reason I don't understand, I can't merge v.next to this branch to fix conflicts.

**Comment by GuillaumeBelz** [comment]
Conflicts resolved

**Review by jared-2016 — APPROVED** [review]
Looks good. I guess we might eventually want to rename `toolkitcpp.pri` to something else not including cpp but that doesn't need to be included in this work.

**Comment by GuillaumeBelz** [comment]
> Looks good. I guess we might eventually want to rename `toolkitcpp.pri` to something else not including cpp but that doesn't need to be included in this work.

Yes, I thought about that. But there is `toolkitwidgets` project. I guess `toolkit` only is not explicit enough.

**Comment by ldanzinger** [comment]
changing the name of `toolkitcpp.pri` would likely have a far reaching impact, probably not worth it.
